components/sup/src/manager/service/mod.rs

```diff
@@ -732,11 +706,6 @@ impl Service {
         Ok(())
     }
 
-    fn run_hook(&mut self, hook: HookType) -> Result<()> {
-        self.hooks.compile(&self.service_group, &self.config);
-        self.hooks.try_run(hook, &self.service_group)
-    }
-
     /// Update our own service rumor with a new configuration from the packages exported
```

Comment:
Perhaps restructuring [this code](https://github.com/habitat-sh/habitat/blob/42c3ff5678cb2c8a18a896138c98708e36da19d8/components/sup/src/manager/service/mod.rs#L418-L421) to ensure `reconfigure` is called not only if `self.config.write()` return `Some(true)` but also if `self.hooks.compile(&self.service_group, &self.config)` indicate a change has been persisted. I guess this would require hashing the compiled hooks like we do the config files.